### DISCUSSION TITLE
Connecting labs to the outside world

### DISCUSSION THREAD

**Discussion by ipspace** [ipspace/netlab#771] [Features]
`docs/roadmap/external-interface.md` (also available @ https://netsim-tools.readthedocs.io/en/dev/roadmap/external-interface.html) describes a simple way to connect a lab link to an external interface. Once we get this to work we could start considering auto-created VLAN interfaces.

Any comments or better ideas?

CC: @ssasso @ddutt @jbemmel @petercrocker

**Comment by jbemmel** [comment]
See https://containerlab.dev/manual/multi-node/#bridging for some related containerlab capabilities. Specifically, it can add any port to an external facing bridge

**Reply by ipspace** [reply]
Thank you, that's exactly what I'm doing to combine _libvirt_ VMs with _clab_ containers, and the same approach will be used to connect external interfaces:

* Mark a link as not being a point-to-point link
* Create a Linux bridge (already done within **clab** provider)
* Start containerlab using that Linux bridge
* Connect external interface to that Linux bridge.

**Comment by ipspace** [comment]
Implemented for libvirt and containerlab (now that it supports macvlan interfaces)
